crates/egui/src/widgets/plot/mod.rs

```diff
@@ -651,10 +737,148 @@ impl Plot {
                 .at_least(min_size.y);
             vec2(width, height)
         };
+        // Determine complete rect of widget.
+        let complete_rect = Rect {
+            min: pos,
+            max: pos + size,
+        };
+        // Next we want to create this layout.
+        // Incides are only examples.
+        //
+        //  +-b-+---------x----------+   +
+        //  |   |      x-Axis 3      |
+        //  c   +--------------------+
+        //  |   |      x-Axis 2      |
+        //  +-+-+--------------------+-+-+
+        //  |y|y|                    |y|y|
+        //  |-|-|                    |-|-|
+        //  |A|A|                    |A|A|
+        // y|x|x|    Plot Window     |x|x|
+        //  |i|i|                    |i|i|
+        //  |s|s|                    |s|s|
+        //  |1|0|                    |2|3|
+        //  +-+-+--------------------+-+-+
+        //      |      x-Axis 0      |   |
+        //      +--------------------+   a
+        //      |      x-Axis 1      |   |
+        //  +   +--------------------+-d-+
+        //
+
+        let mut plot_rect: Rect = {
+            // find dimensions of axis labels
+            // for a, b, c, d meanings see picture
+            let mut a = 0.0;
+            let mut b = 0.0;
+            let mut c = 0.0;
+            let mut d = 0.0;
+            if show_axes.x {
+                for cfg in &x_axes {
+                    match cfg.placement {
+                        axis::Placement::Default => {
+                            a += cfg.thickness();
+                        }
+                        axis::Placement::Opposite => {
+                            c += cfg.thickness();
+                        }
+                    }
+                }
+            }
+            if show_axes.y {
+                for cfg in &y_axes {
+                    match cfg.placement {
+                        axis::Placement::Default => {
+                            b += cfg.thickness();
+                        }
+                        axis::Placement::Opposite => {
+                            d += cfg.thickness();
+                        }
+                    }
+                }
+            }
+
+            // determine plot rectangle
+            Rect {
+                min: complete_rect.min + Vec2::new(b, c),
+                max: complete_rect.max - Vec2::new(d, a),
+            }
+        };
+        let mut x_axis_widgets = Vec::<XAxisWidget>::new();
+        let mut y_axis_widgets = Vec::<YAxisWidget>::new();
+        {
+            // determine absolute rectangle for each axis label widget
+            // widget cnt per border of plot in order left, top, right, bottom
+            struct WidgetCnt {
```

Comment:
```suggestion
            struct NumWidgets {
```

is a name more consistent with the rest of the code base